Adds ls troubleshooting tips (#5266)
dbt-labs/docs.getdbt.com: website/docs/reference/node-selection/syntax.md

consider using a [yaml selector](/reference/node-selection/yaml-selectors). You can use a predefined definition with the `--selector` flag.
Note that when you're using `--selector`, most other flags (namely `--select` and `--exclude`) will be ignored.

### Troubleshoot with the `ls` command

Constructing and debugging your selection syntax can be challenging.  To get a "preview" of what will be selected, we recommend using the [`list` command](/reference/commands/list).  This command, when combined with your selection syntax, will output a list of the nodes that meet that selection criteria.  The `dbt ls` command supports all types of selection syntax arguments, for example:

> ```bash

Reviewer: hey @will-sargent-dbtlabs this is a great pr and what a wealth of knowledge 🙏 - thank you for sharing! I've tweaked it so it sits above the discourse community header and also added context as to what each command does - can you verify if that's technically right? i wasn't sure about the 2nd and 4th examples

> dbt ls --select "path.to.my.models" # Lists all models in a specific directory.

Reviewer: should it be this?
```suggestion
dbt ls --select "path/to/my/models" # Lists all models in a specific directory.
```

Author: I copied that from another example on the page. But I agree that slashes makes more sense

Author: I don't think dots actually work AFAIK
I got this from the "Examples of select flag" block up the page

Reviewer: awesome, thank yoU! i'll try to find the other one and update it too! 🙏

dbt ls --select "source_status:fresher+" # Shows sources updated since the last dbt source freshness run.
dbt ls --select state:modified+ # Displays nodes modified in comparison to a previous state.

> dbt ls --select "result:<status>+ state:modified+ --defer --state ./<dbt-artifact-path>" # Lists nodes that match certain result statuses and are modified, which can help with troubleshooting.

Reviewer: ```suggestion
dbt ls --select "result:STATUS+ state:modified+ --defer --state ./dbt-artifact-path" # Lists nodes that match certain result statuses and are modified, which can help with troubleshooting.
```

> dbt ls --select "result:<status>+ state:modified+ --defer --state ./<dbt-artifact-path>" # Lists nodes that match certain result statuses and are modified, which can help with troubleshooting.

Author: This is a core team question on if this is valid, because I lifted this from below as well..
Either way, if we removed it, no big deal as it was just a sample example of something complicated

Author: Maybe that is a placeholder for SUCCESS or FAILED?

Author: Yep:
result:error	✅	✅	✅	✅
result:success	✅	✅	✅	
result:skipped	✅		✅	✅
result:fail				✅
result:warn				✅
result:pass

Author: So I think it's probably ok Maybe we need to tweak the table itself too..

Author: Added a commit to do that

> dbt ls --select "result:<status>+ state:modified+ --defer --state ./<dbt-artifact-path>" # Lists nodes that match certain result statuses and are modified, which can help with troubleshooting.

Reviewer: ```suggestion
dbt ls --select "result:<status>+ state:modified+ --defer --state ./<dbt-artifact-path>" # Lists nodes that match certain [result statuses](/reference/node-selection/syntax#the-result-status) and are modified.
```

<Snippet path="discourse-help-feed-header" />
<DiscourseHelpFeed tags="node-selection"/>

> ### Troubleshooting tip: Use the `ls` command!

Reviewer: ```suggestion
```
